### PR TITLE
Dev mw

### PR DESCRIPTION
**Pull request by wiederm**
## Description
Adding:
- mining minima approach to the tautomer class,
- droplet generation
- COM restraints 

## Todos
Notable points that this PR has either accomplished or will accomplish.
  -  generating conformations
  - sorting and filtering conformations
  - boltzmann weighting of energies
  - generate a droplet with a given diameter and the ligand in the middle
  - sets a COM restraint on the ligand to keep it in the center
## Status
- [x] Ready to go

### CODE REVIEW

File: neutromeratio/restraints.py [hunk 1]
```diff
@@ -34,23 +77,23 @@ def __init__(self, sigma:unit.Quantity, atom_i_idx:int, atom_j_idx:int, atoms:st
...
-        self.k = k.value_in_unit((unit.kilo * unit.joule) / ((unit.angstrom **2) * unit.mole))
-        self.device = device
+        super().__init__(sigma, active_at_lambda)
         self.atom_i_element = atoms[atom_i_idx]
         self.atom_j_element = atoms[atom_j_idx]
         self.atom_i_idx = atom_i_idx
         self.atom_j_idx = atom_j_idx
         self.mean_bond_length = (bond_length_dict[frozenset([self.atom_i_element, self.atom_j_element])]).value_in_unit(unit.angstrom)
-        self.active_at_lambda = active_at_lambda
         self.upper_bound = self.mean_bond_length + 0.2
```
Comment (maxentile) [inline neutromeratio/restraints.py]: Leave a `# TODO: ` comment here noting that the width of the restraint is hardcoded

File: neutromeratio/tautomers.py [hunk 2]
```diff
@@ -305,4 +433,164 @@ def _generate_hybrid_structure(self, ligand_atoms:str, ase_mol:Atoms, ligand_coo
...
+                    if i >= j:
+                        continue
+                    rmsd[i, j] = AllChem.GetBestRMS(mol, mol, ref_conf.GetId(),
+                                                    fit_conf.GetId())
+                    rmsd[j, i] = rmsd[i, j]
+            return rmsd
+
 
+        def calculate_weighted_energy(e_list):
+            #G = -RT Σ ln exp(-G/RT)
```
Comment (maxentile) [inline neutromeratio/tautomers.py]: Are `Σ` and `ln` transposed here?

File: neutromeratio/tests/test_neutromeratio.py [hunk 3]
```diff
@@ -315,61 +311,52 @@ def test_restraint_with_alchemicalANI():
...
-
-    x = energy_function.calculate_energy(x0, lambda_value=0.0)
-    assert(is_quantity_close(x, -216698.911373172* unit.kilocalorie_per_mole, rtol=1e-9))
 
     # test flat_bottom_restraint for lambda = 0.0
-    energy_function.flat_bottom_restraint  = True
-    energy_function.harmonic_restraint  = False
-    energy_function.use_pure_ani1ccx = False
+    r = []
+    restrain1 = neutromeratio.restraints.FlatBottomRestraint(sigma=0.1 * unit.angstrom, atom_i_idx=6, atom_j_idx=7, atoms=atoms, active_at_lambda=0)
```
Comment (maxentile) [inline neutromeratio/tests/test_neutromeratio.py]: Nice! This is much clearer!

File: neutromeratio/tests/test_neutromeratio.py [hunk 4]
```diff
@@ -315,61 +311,52 @@ def test_restraint_with_alchemicalANI():
...
+    restrain2 = neutromeratio.restraints.FlatBottomRestraint(sigma=0.1 * unit.angstrom, atom_i_idx=6, atom_j_idx=7, atoms=atoms, active_at_lambda=1)
+    restrain3 = neutromeratio.restraints.FlatBottomRestraint(sigma=0.1 * unit.angstrom, atom_i_idx=6, atom_j_idx=7, atoms=atoms, active_at_lambda=-1)
```
Comment (maxentile) [inline neutromeratio/tests/test_neutromeratio.py]: Hmm, could we clarify usage convention of lambda=-1,0,+1?

File: neutromeratio/plotting.py [hunk 5]
```diff
@@ -64,20 +62,18 @@ def plot_correlation_analysis(
...
     ax.set_ylim([-22, 22])
-    #plt.xticks(np.arange(-10, 10, 2.0), fontsize=8)
+    #plt.xticks(np.arange(-10, 10, 2.0), fontsize=15)
     handles, labels = plt.gca().get_legend_handles_labels()
     by_label = OrderedDict(zip(labels, handles))
     plt.legend(by_label.values(), by_label.keys(), loc='lower right')
     plt.tight_layout()
-    
     plt.subplots_adjust(bottom=0.3), #left=1.3, right=0.3) 
+    return ax
```
[the remark this answers is not in the document]
Comment (wiederm) [reply neutromeratio/plotting.py]: interestingly, `ax.axis('equal')` overwrites `ax.set_xlim()` and `ax.set_ylim()`. the method that does the job and still respects x/ylimi is `ax.set_aspect('equal', 'box')`! Added!

[the remark this answers is not in the document]
Comment (maxentile) [reply neutromeratio/plotting.py]: Ahh, good catch -- thanks!!

**Pull request merged**